library/cpp/include/wavemap/core/utils/profile/resource_monitor.h

```diff
@@ -0,0 +1,150 @@
+#ifndef WAVEMAP_CORE_UTILS_PROFILE_RESOURCE_MONITOR_H_
```

Comment:
Couldn't this just use the Stopwatch class internally? Seems like they are very similar in some aspects.

Reply:
They're on different clocks (or at least not guaranteed to be on the same one). Stopwatch uses chrono's steady_clock, whereas ResourceMonitor gets its wall time from a POSIX call.

Reply:
👍